packages/ai/core/generate-speech/generate-speech.ts

```diff
@@ -0,0 +1,159 @@
+import { JSONValue, SpeechModelV1 } from '@ai-sdk/provider';
+import { NoSpeechGeneratedError } from '../../errors/no-speech-generated-error';
+import { prepareRetries } from '../prompt/prepare-retries';
+import { ProviderOptions } from '../types/provider-metadata';
+import { SpeechWarning } from '../types/speech-model';
+import { SpeechModelResponseMetadata } from '../types/speech-model-response-metadata';
+import { SpeechResult } from './generate-speech-result';
+import {
+  DefaultGeneratedFile,
+  GeneratedFile,
+} from '../generate-text/generated-file';
+import {
+  audioMimeTypeSignatures,
+  detectMimeType,
+} from '../util/detect-mimetype';
+
+/**
+Generates speech audio using a speech model.
+
+@param model - The speech model to use.
+@param text - The text to convert to speech.
+@param voice - The voice to use for speech generation.
+@param outputMediaType - The output media type to use for speech generation.
+@param instructions - Instructions for the speech generation e.g. "Speak in a slow and steady tone".
+@param speed - The speed of the speech generation.
+@param providerOptions - Additional provider-specific options that are passed through to the provider
+as body parameters.
+@param maxRetries - Maximum number of retries. Set to 0 to disable retries. Default: 2.
+@param abortSignal - An optional abort signal that can be used to cancel the call.
+@param headers - Additional HTTP headers to be sent with the request. Only applicable for HTTP-based providers.
+
+@returns A result object that contains the generated audio data.
+ */
+export async function generateSpeech({
+  model,
+  text,
+  voice,
+  outputMediaType,
+  instructions,
+  speed,
+  providerOptions = {},
+  maxRetries: maxRetriesArg,
+  abortSignal,
+  headers,
+}: {
+  /**
+The speech model to use.
+     */
+  model: SpeechModelV1;
+
+  /**
+The text to convert to speech.
+   */
+  text: string;
+
+  /**
+The voice to use for speech generation.
+   */
+  voice?: string;
+
+  /**
+The output media type to use for speech generation.
+   */
```

Comment:
Add examples. We might want to change this, media types would be `audio/mp3` etc but i think we want `mp3` only

Reply:
also we could have nicer auto complete e.g. `'audio/wav' | 'audio/mp3' | ({} & string)`